app/src/main/java/com/siba/searchmvvmpractice/local/database/SearchTermDatabase.kt

```diff
@@ -0,0 +1,34 @@
+package com.siba.searchmvvmpractice.local.database
+
+import android.content.Context
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+import com.siba.searchmvvmpractice.local.dao.SearchTermDao
+import com.siba.searchmvvmpractice.local.entity.RecentSearchTerm
+
+@Database(entities = [RecentSearchTerm::class], version = 1)
+abstract class SearchTermDatabase : RoomDatabase() {
+    abstract val searchTermDao: SearchTermDao
+
+    companion object {
+        @Volatile
+        private var INSTANCE: SearchTermDatabase? = null
+
+        fun getInstance(context: Context): SearchTermDatabase {
+            synchronized(this) {
+                var instance = INSTANCE
+
+                if (instance == null) {
+                    instance = Room.databaseBuilder(
+                        context.applicationContext,
+                        SearchTermDatabase::class.java,
+                        "search_keyword_history_database2"
+                    ).build()
+                    INSTANCE = instance
+                }
+                return instance
```

Comment:
이 부분에 대한 코드인데 사실 구글에서 하라는 대로긴 하지만 너무 깊게 depth가 들어가는 부분이라 나는 이 부분은 좀 자바 같다고 생각하는데 어떻게 생각해?
depth를 줄이고 가독성 있게 만드는 방법이 있지 않을까?

그리고 약간 return 값이 instance인데 내부 변수인 이 친구를 던지는 이유는 뭘까..? 따로 변수를 만들지 않아도 충분히 가능한 부분 같아

Reply:
```suggestion
        fun getInstance(context: Context): SearchTermDatabase = INSTANCE ?: synchronized(this) {
    INSTANCE ?: Room.databaseBuilder(
        context.applicationContext,
        SearchTermDatabase::class.java,
        "search_keyword_history_database2"
    ).build().apply {
        INSTANCE = this
    }
}
```
이런 식으로 하면 기존 코드보다 뎁스도 줄이고 가독성도 해치지 않는다고 생각해

Reply:
해당 함수가 SearchTermDatabase 타입의 INSTANCE를 던져주는데 만약 null이라면 동기화 블럭에 들어가서 Room 데이터베이스 빌더로 데이터 베이스 만들어주고 그걸 INSTANCE에 apply하는거지

그리고 기존 코드랑 똑같은데 다르게 작성해서 indent 를 줄여본거고

큰 차이점이 있다면 미리 우리가 선언해둔 인스턴스에 값이 들어있다면 동기화 블럭을 거치지 않아도 해당 인스턴스를 던져주는 느낌?
난 기존코드를 생각했을때 INSTANCE가 있던 없던 동기화 블럭에 간다고 생각해서
굳이 이미 INSTANCE가 있을때도 멀티스레드 세이프티하게 인스턴스를 던져줄 필요가 없다고 생각했어 어짜피 그땐 동일한 INSTANCE를 가져가니까